### PR TITLE
my hw array list

### PR DESCRIPTION
**Pull request by VolodymyrKachan**

### CODE REVIEW

File: src/main/java/core/basesyntax/ArrayList.java [hunk 1]
```diff
@@ -1,48 +1,102 @@
...
 
     @Override
     public void set(T value, int index) {
-
+        if (index < size && index >= 0) {
+            defaultArray[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("OutOfBoundsArrayList: " + index);
+        }
     }
```
Comment (victoriadobryden) [inline src/main/java/core/basesyntax/ArrayList.java:51]: you have function checkIndex() for this, let's use it here

File: src/main/java/core/basesyntax/ArrayList.java [hunk 2]
```diff
@@ -1,48 +1,102 @@
...
+            if (defaultArray[i] != null ? defaultArray[i].equals(element)
+                    : defaultArray[i] == element) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("There is no such element in ArrayList");
+    }
+
+    private void checkIndex(int index) {
+        if (index > size || index < 0) {
```
Comment (victoriadobryden) [inline src/main/java/core/basesyntax/ArrayList.java:74]: `if (index >= size || index < 0) {`

**Review by victoriadobryden — CHANGES_REQUESTED** [review]

**Review by victoriadobryden — APPROVED** [review]

File: src/main/java/core/basesyntax/ArrayList.java [hunk 3]
```diff
@@ -1,48 +1,95 @@
...
+        throw new NoSuchElementException("There is no such element in ArrayList");
+    }
+
+    private void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index not valid: " + index);
+        }
+    }
+
+    private void checkArrayPlus() {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java:78]:
```suggestion
    private void resizeArray() {
```

File: src/main/java/core/basesyntax/ArrayList.java [hunk 4]
```diff
@@ -1,48 +1,95 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double ARRAY_MAGNIFICATION = 1.5;
+    private int size;
+    private T[] defaultArray;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java:9]:
```suggestion
    private T[] elements/data;
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]
great work

File: src/main/java/core/basesyntax/ArrayList.java [hunk 5]
```diff
@@ -1,48 +1,95 @@
...
+            throw new ArrayListIndexOutOfBoundsException("Index not valid: " + index);
+        }
+    }
+
+    private void resizeArray() {
+        if (elements.length == size) {
+            T[] newArray = (T[]) new Object[(int) (elements.length * ARRAY_MAGNIFICATION)];
+            System.arraycopy(elements, 0, newArray, 0, size);
+            elements = newArray;
+        }
```
Comment (Noct2000) [inline src/main/java/core/basesyntax/ArrayList.java:83]: private methods must be after public

**Review by Noct2000 — CHANGES_REQUESTED** [review]

**Review by Noct2000 — APPROVED** [review]

**Pull request closed without merging**